### DISCUSSION TITLE
Add support for _TZ3000_b3mgfu0d, TS004F

### DISCUSSION THREAD

**Discussion by dhc25** [zigpy/zha-device-handlers#2853] [General]
Sorry if this is the wrong process to seek an additional model to be added to the existing quirk. (New to github)

Can you please add this Tuya model to the existing quirk for tuya/ts004f.py, in the TuyaSmartRemote004FDMS models.

_TZ3000_b3mgfu0d
TS004F

I have tested this with a custom quirk, and this model works exactly the same as _TZ3000_czuyt8lz

I also note that neither _TZ3000_czuyt8lz nor _TZ3000_b3mgfu0d report battery levels.
They do not report battery levels even when paired to a Tuya Zigbee hub, so battery levels in HA are always wrong.
They are USB powered.

Can the reporting of battery levels in HA be removed?

Thank you

**Comment by MattWestb** [comment]
Can you posting one photo of the device or one working web link ?

**Comment by dhc25** [comment]
https://www.amazon.co.uk/Candeo-Wireless-requires-interchangeable-rechargeable-White/dp/B09SN5QS2M

This is _TZ3000_czuyt8lz

The device I have in hand (_TZ3000_b3mgfu0d) is exactly the same, from manufacturer

**Comment by MattWestb** [comment]
The black looks very nice !!!!
Can you also posting the diagnostic from the device device-card then need to see if they have flagging the firmware as battery powered or not.
The original "_TZ3000_xabckq1v" is reporting battery then pressing 2 and also self if being reconfigured OK.

**Comment by dhc25** [comment]
I'm not sure if this is what you're looking for...




{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2023.12.3",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.11.6",
    "docker": true,
    "arch": "aarch64",
    "timezone": "Europe/London",
    "os_name": "Linux",
    "os_version": "6.1.58-haos-raspi",
    "supervisor": "2023.12.0",
    "host_os": "Home Assistant OS 11.2",
    "docker_version": "24.0.7",
    "chassis": "embedded",
    "run_as_root": true
  },
  "custom_components": {
    "tuya_local": {
      "version": "2023.12.0",
      "requirements": [
        "tinytuya==1.13.1"
      ]
    },
    "hacs": {
      "version": "1.33.0",
      "requirements": [
        "aiogithubapi>=22.10.1"
      ]
    },
    "pirateweather": {
      "version": "1.2.3",
      "requirements": [
        "python-forecastio==1.4.0"
      ]
    },
    "skyq": {
      "version": "v2.11.6",
      "requirements": [
        "pyskyqremote==0.3.25"
      ]
    },
    "ui_lovelace_minimalist": {
      "version": "v1.3.9",
      "requirements": [
        "aiofiles==0.8.0",
        "aiogithubapi>=22.2.4"
      ]
    },
    "audiconnect": {
      "version": "2021.1.1",
      "requirements": [
        "beautifulsoup4"
      ]
    },
    "ukho_tides": {
      "version": "0.1.1",
      "requirements": [
        "ukhotides==0.1.1"
      ]
    },
    "meross_cloud": {
      "version": "1.2.10",
      "requirements": [
        "meross_iot==0.4.6.0rc2"
      ]
    },
    "nest_protect": {
      "version": "0.4.0b1",
      "requirements": []
    },
    "roborock": {
      "version": "1.0.13",
      "requirements": [
        "python-roborock==0.36.0",
        "ical==6.1.0",
        "dacite==1.8.0"
      ]
    },
    "localtuya": {
      "version": "5.2.1",
      "requirements": []
    },
    "anniversaries": {
      "version": "5.2.0",
      "requirements": [
        "python-dateutil>=2.8.1",
        "integrationhelper>=0.2.2",
        "voluptuous>=0.12.1"
      ]
    },
    "hubitat": {
      "version": "0.9.8",
      "requirements": []
    },
    "smartthinq_sensors": {
      "version": "0.35.5",
      "requirements": [
        "pycountry>=20.7.3",
        "xmltodict>=0.12.0",
        "charset_normalizer>=2.0.0"
      ]
    },
    "browser_mod": {
      "version": "2.3.0",
      "requirements": []
    },
    "google_fit": {
      "version": "v3.0.0",
      "requirements": [
        "oauth2client==4.1.3",
        "google-api-python-client==2.82.0"
      ]
    },
    "zha_toolkit": {
      "version": "v1.1.8",
      "requirements": [
        "pytz"
      ]
    }
  },
  "integration_manifest": {
    "domain": "zha",
    "name": "Zigbee Home Automation",
    "after_dependencies": [
      "onboarding",
      "usb"
    ],
    "codeowners": [
      "@dmulcahey",
      "@adminiuga",
      "@puddly",
      "@TheJulianJES"
    ],
    "config_flow": true,
    "dependencies": [
      "file_upload"
    ],
    "documentation": "https://www.home-assistant.io/integrations/zha",
    "iot_class": "local_polling",
    "loggers": [
      "aiosqlite",
      "bellows",
      "crccheck",
      "pure_pcapy3",
      "zhaquirks",
      "zigpy",
      "zigpy_deconz",
      "zigpy_xbee",
      "zigpy_zigate",
      "zigpy_znp",
      "universal_silabs_flasher"
    ],
    "requirements": [
      "bellows==0.37.3",
      "pyserial==3.5",
      "pyserial-asyncio==0.6",
      "zha-quirks==0.0.107",
      "zigpy-deconz==0.22.2",
      "zigpy==0.60.1",
      "zigpy-xbee==0.20.1",
      "zigpy-zigate==0.12.0",
      "zigpy-znp==0.12.0",
      "universal-silabs-flasher==0.0.15",
      "pyserial-asyncio-fast==0.11"
    ],
    "usb": [
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*2652*",
        "known_devices": [
          "slae.sh cc2652rb stick"
        ]
      },
      {
        "vid": "1A86",
        "pid": "55D4",
        "description": "*sonoff*plus*",
        "known_devices": [
          "sonoff zigbee dongle plus v2"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*sonoff*plus*",
        "known_devices": [
          "sonoff zigbee dongle plus"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*tubeszb*",
        "known_devices": [
          "TubesZB Coordinator"
        ]
      },
      {
        "vid": "1A86",
        "pid": "7523",
        "description": "*tubeszb*",
        "known_devices": [
          "TubesZB Coordinator"
        ]
      },
      {
        "vid": "1A86",
        "pid": "7523",
        "description": "*zigstar*",
        "known_devices": [
          "ZigStar Coordinators"
        ]
      },
      {
        "vid": "1CF1",
        "pid": "0030",
        "description": "*conbee*",
        "known_devices": [
          "Conbee II"
        ]
      },
      {
        "vid": "0403",
        "pid": "6015",
        "description": "*conbee*",
        "known_devices": [
          "Conbee III"
        ]
      },
      {
        "vid": "10C4",
        "pid": "8A2A",
        "description": "*zigbee*",
        "known_devices": [
          "Nortek HUSBZB-1"
        ]
      },
      {
        "vid": "0403",
        "pid": "6015",
        "description": "*zigate*",
        "known_devices": [
          "ZiGate+"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*zigate*",
        "known_devices": [
          "ZiGate"
        ]
      },
      {
        "vid": "10C4",
        "pid": "8B34",
        "description": "*bv 2010/10*",
        "known_devices": [
          "Bitron Video AV2010/10"
        ]
      }
    ],
    "zeroconf": [
      {
        "type": "_esphomelib._tcp.local.",
        "name": "tube*"
      },
      {
        "type": "_zigate-zigbee-gateway._tcp.local.",
        "name": "*zigate*"
      },
      {
        "type": "_zigstar_gw._tcp.local.",
        "name": "*zigstar*"
      },
      {
        "type": "_uzg-01._tcp.local.",
        "name": "uzg-01*"
      },
      {
        "type": "_slzb-06._tcp.local.",
        "name": "slzb-06*"
      }
    ],
    "is_built_in": true
  },
  "data": {
    "ieee": "**REDACTED**",
    "nwk": 15936,
    "manufacturer": "_TZ3000_b3mgfu0d",
    "model": "TS004F",
    "name": "_TZ3000_b3mgfu0d TS004F",
    "quirk_applied": true,
    "quirk_class": "ts004f.TuyaSmartRemote004FDMS",
    "quirk_id": null,
    "manufacturer_code": 4417,
    "power_source": "Battery or Unknown",
    "lqi": 232,
    "rssi": -42,
    "last_seen": "2023-12-21T13:13:21",
    "available": true,
    "device_type": "EndDevice",
    "signature": {
      "node_descriptor": "NodeDescriptor(logical_type=<LogicalType.EndDevice: 2>, complex_descriptor_available=0, user_descriptor_available=0, reserved=0, aps_flags=0, frequency_band=<FrequencyBand.Freq2400MHz: 8>, mac_capability_flags=<MACCapabilityFlags.AllocateAddress: 128>, manufacturer_code=4417, maximum_buffer_size=66, maximum_incoming_transfer_size=66, server_mask=10752, maximum_outgoing_transfer_size=66, descriptor_capability_field=<DescriptorCapability.NONE: 0>, *allocate_address=True, *is_alternate_pan_coordinator=False, *is_coordinator=False, *is_end_device=True, *is_full_function_device=False, *is_mains_powered=False, *is_receiver_on_when_idle=False, *is_router=False, *is_security_capable=False)",
      "endpoints": {
        "1": {
          "profile_id": "0x0104",
          "device_type": "0x0820",
          "input_clusters": [
            "0x0000",
            "0x0001",
            "0x0003",
            "0x0004",
            "0x1000"
          ],
          "output_clusters": [
            "0x0003",
            "0x0004",
            "0x0005",
            "0x0006",
            "0x0008",
            "0x000a",
            "0x0019",
            "0x0300",
            "0x1000"
          ]
        },
        "2": {
          "profile_id": "0x0104",
          "device_type": "0x0820",
          "input_clusters": [
            "0x0006"
          ],
          "output_clusters": []
        },
        "3": {
          "profile_id": "0x0104",
          "device_type": "0x0820",
          "input_clusters": [
            "0x0006"
          ],
          "output_clusters": []
        },
        "4": {
          "profile_id": "0x0104",
          "device_type": "0x0820",
          "input_clusters": [
            "0x0006"
          ],
          "output_clusters": []
        }
      },
      "manufacturer": "_TZ3000_b3mgfu0d",
      "model": "TS004F"
    },
    "active_coordinator": false,
    "entities": [
      {
        "entity_id": "button.tz3000_b3mgfu0d_ts004f_identify",
        "name": "_TZ3000_b3mgfu0d TS004F"
      },
      {
        "entity_id": "sensor.tz3000_b3mgfu0d_ts004f_battery",
        "name": "_TZ3000_b3mgfu0d TS004F"
      }
    ],
    "neighbors": [],
    "routes": [],
    "endpoint_names": [
      {
        "name": "NON_COLOR_CONTROLLER"
      },
      {
        "name": "NON_COLOR_CONTROLLER"
      },
      {
        "name": "NON_COLOR_CONTROLLER"
      },
      {
        "name": "NON_COLOR_CONTROLLER"
      }
    ],
    "user_given_name": null,
    "device_reg_id": "c7cd29d8ceaf95fdf70532b8355f4634",
    "area_id": null,
    "cluster_details": {
      "1": {
        "device_type": {
          "name": "NON_COLOR_CONTROLLER",
          "id": 2080
        },
        "profile_id": 260,
        "in_clusters": {
          "0x0000": {
            "endpoint_attribute": "basic",
            "attributes": {
              "0x0001": {
                "attribute_name": "app_version",
                "value": 66
              },
              "0x0004": {
                "attribute_name": "manufacturer",
                "value": "_TZ3000_b3mgfu0d"
              },
              "0x0005": {
                "attribute_name": "model",
                "value": "TS004F"
              },
              "0x0007": {
                "attribute_name": "power_source",
                "value": 3
              },
              "0xfffe": {
                "attribute_name": "reporting_status",
                "value": 0
              },
              "0x0000": {
                "attribute_name": "zcl_version",
                "value": 3
              }
            },
            "unsupported_attributes": {}
          },
          "0x0001": {
            "endpoint_attribute": "power",
            "attributes": {
              "0x0021": {
                "attribute_name": "battery_percentage_remaining",
                "value": 2
              },
              "0x0020": {
                "attribute_name": "battery_voltage",
                "value": 25
              }
            },
            "unsupported_attributes": {
              "0x0031": {
                "attribute_name": "battery_size"
              },
              "0x0033": {
                "attribute_name": "battery_quantity"
              }
            }
          },
          "0x0003": {
            "endpoint_attribute": "identify",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0004": {
            "endpoint_attribute": "groups",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x1000": {
            "endpoint_attribute": "lightlink",
            "attributes": {},
            "unsupported_attributes": {}
          }
        },
        "out_clusters": {
          "0x0019": {
            "endpoint_attribute": "ota",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x000a": {
            "endpoint_attribute": "time",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0003": {
            "endpoint_attribute": "identify",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0004": {
            "endpoint_attribute": "groups",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0005": {
            "endpoint_attribute": "scenes",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0006": {
            "endpoint_attribute": "TS004X_cluster",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0008": {
            "endpoint_attribute": "level",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0300": {
            "endpoint_attribute": "light_color",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x1000": {
            "endpoint_attribute": "lightlink",
            "attributes": {},
            "unsupported_attributes": {}
          }
        }
      },
      "2": {
        "device_type": {
          "name": "NON_COLOR_CONTROLLER",
          "id": 2080
        },
        "profile_id": 260,
        "in_clusters": {
          "0x0006": {
            "endpoint_attribute": "TS004X_cluster",
            "attributes": {},
            "unsupported_attributes": {}
          }
        },
        "out_clusters": {}
      },
      "3": {
        "device_type": {
          "name": "NON_COLOR_CONTROLLER",
          "id": 2080
        },
        "profile_id": 260,
        "in_clusters": {
          "0x0006": {
            "endpoint_attribute": "TS004X_cluster",
            "attributes": {},
            "unsupported_attributes": {}
          }
        },
        "out_clusters": {}
      },
      "4": {
        "device_type": {
          "name": "NON_COLOR_CONTROLLER",
          "id": 2080
        },
        "profile_id": 260,
        "in_clusters": {
          "0x0006": {
            "endpoint_attribute": "TS004X_cluster",
            "attributes": {},
            "unsupported_attributes": {}
          }
        },
        "out_clusters": {}
      }
    }
  }
}



I think the the _TZ3000_xabckq1v is battery powered (coin battery), but the 2 devices I am referring to are USB charged batteries.

**Reply by MattWestb** [reply]
Very true !!
From the firmware is saying:
```
*is_end_device=True, 
*is_mains_powered=False, 
*is_receiver_on_when_idle=False,
```
So its not mains powered (AC or UPS).

Can you reading the `power_source (id: 0x0007` on the basic cluster (manage Zigbee device from the device card on the optiony ... right from reconfigure) and see what is saying ( wak it up befor sending somthing to it).

The "old" is saying `PowerSource.Battery`.

**Comment by dhc25** [comment]
This says the same - "PowerSource.Battery"

**Reply by MattWestb** [reply]
Then i recommending Disabl the battery sensor on the device card and it shall dispersing from the HA GUI.
Its possible deleting it in the quirk but then its need one new device class for not doing the same with the other 2 devices that is using the same quirk class.

**Reply by dhc25** [reply]
Ok - that makes sense - thanks for your help.

So - 2 questions - 
1. How should I go about submitting this reference _TZ3000_b3mgfu0d to be added to existing quirk in HA?
2. For my reference, how would I delete the battery in the quirk, if I was to continue using a custom quirk?

Thank you

**Reply by MattWestb** [reply]
Best is doing one PR and then its being integrated in the next HA release.
Open the file in GITHUB and then editing it and then all is doing post the changes and hope the maintainer is merging it (be careful with formatting for not braking the code).
If some more advanced user like they can doing more changes and deleting the battery in one new device class and adding your devie to it.

**Reply by dhc25** [reply]
Thanks for your guidance
